### PR TITLE
JAMES-3168: distributed EchoMethod integration test

### PR DESCRIPTION
**Pull request by nvduc91**
distributed EchoMethod integration test

### CODE REVIEW

File: server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/pom.xml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>jmap-rfc8621-integration-tests</artifactId>
+        <groupId>org.apache.james</groupId>
+        <version>3.6.0-SNAPSHOT</version>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>distributed-jmap-rfc8621-integration-tests</artifactId>
+
```
Comment (chibenwa) [inline server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/pom.xml:31]: Having a nice description here would be nice

File: server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/pom.xml [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            <artifactId>james-server-guice-jmap</artifactId>
+            <type>test-jar</type>
+        </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-guice-common</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-memory-guice</artifactId>
```
Comment (chibenwa) [inline server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/pom.xml:110]: Why is this needed here?

File: server/protocols/jmap-rfc8621-integration-tests/jmap-rfc8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EchoMethodContract.scala [hunk 3]
```diff
@@ -149,7 +152,8 @@ trait EchoMethodContract {
   }
 
   @Test
-  def echoMethodShouldRespondWithRFC8621VersionAndUnsupportedMethod(): Unit = {
+  @Category(Array(classOf[BasicFeature]))
```
Comment (chibenwa) [inline server/protocols/jmap-rfc8621-integration-tests/jmap-rfc8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EchoMethodContract.scala:155]: IMO we should test data validation (unsupported method) only on memory product

File: server/protocols/jmap-rfc8621-integration-tests/pom.xml [hunk 4]
```diff
@@ -60,6 +61,12 @@
                 <artifactId>jmap-rfc8621-integration-tests-common</artifactId>
                 <version>${project.version}</version>
             </dependency>
+            <dependency>
+                <groupId>${project.groupId}</groupId>
+                <artifactId>james-server-jmap-draft-integration-testing</artifactId>
```
Comment (chibenwa) [inline server/protocols/jmap-rfc8621-integration-tests/pom.xml:66]: We should not depend on draft something. Please duplicate the annotation.

File: server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/pom.xml (hunk 2 above)
Comment (nvduc91) [reply server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/pom.xml:110]: Yes, it should not needed.

File: server/protocols/jmap-rfc8621-integration-tests/pom.xml (hunk 4 above)
Comment (nvduc91) [reply server/protocols/jmap-rfc8621-integration-tests/pom.xml:66]: Junit 5 change from `Category` to `Tag` and it just need tag's name, we dont need create interface anymore.
We just need remove it and duplicate nothing.

File: server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/pom.xml [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    <parent>
+        <artifactId>jmap-rfc8621-integration-tests</artifactId>
+        <groupId>org.apache.james</groupId>
+        <version>3.6.0-SNAPSHOT</version>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>distributed-jmap-rfc8621-integration-tests</artifactId>
+    <description>Distributed Integration testing for Jmap rfc8621</description>
+    <name>Apache James :: Server :: JMAP RFC-8621 :: Distributed Integration testing</name>
```
Comment (mbaechler) [inline server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/pom.xml:32]:
```suggestion
    <name>Apache James :: Server :: JMAP RFC-8621 :: Distributed Integration Testing</name>
```

File: server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/pom.xml [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>apache-james-backends-cassandra</artifactId>
+            <type>test-jar</type>
+        </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>apache-james-backends-rabbitmq</artifactId>
+            <type>test-jar</type>
+            <scope>test</scope>
```
Comment (mbaechler) [inline server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/pom.xml:48]: why other dependencies are in compile scope and this one is in test scope?

File: server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedEchoMethodTest.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+    @RegisterExtension
+    static JamesServerExtension testExtension = new JamesServerBuilder()
+        .extension(new DockerElasticSearchExtension())
+        .extension(new CassandraExtension())
+        .extension(new RabbitMQExtension())
+        .extension(new AwsS3BlobStoreExtension())
+        .server(configuration -> GuiceJamesServer.forConfiguration(configuration)
+            .combineWith(CassandraRabbitMQJamesServerMain.MODULES)
+            .overrideWith(TestJMAPServerModule.maximumMessages(LIMIT_TO_20_MESSAGES)))
```
Comment (mbaechler) [inline server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedEchoMethodTest.java:46]: any reason we keep that max message setting?

File: server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/imapserver.xml [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+
+<imapservers>
+    <imapserver enabled="true">
+        <jmxName>imapserver</jmxName>
```
Comment (mbaechler) [inline server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/imapserver.xml:25]: are we using imap at all in this project?
on two different ports?

File: server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/lmtpserver.xml [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+  specific language governing permissions and limitations      
+  under the License.                                           
+ -->
+
+<lmtpservers>
+
+    <lmtpserver enabled="true">
+        <jmxName>lmtpserver</jmxName>
+        <!-- LMTP should not be reachable from outside your network so bind it to loopback-->
+        <bind>0.0.0.0:0</bind>
```
Comment (mbaechler) [inline server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/lmtpserver.xml:26]: are we using lmtp at all?

File: server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/pop3server.xml [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  KIND, either express or implied.  See the License for the    
+  specific language governing permissions and limitations      
+  under the License.                                           
+ -->
+
+
+<pop3servers>
+    <pop3server enabled="true">
+        <jmxName>pop3server</jmxName>
+        <bind>0.0.0.0:0</bind>
```
Comment (mbaechler) [inline server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/pop3server.xml:25]: are we using pop3 at all?

File: server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/smtpserver.xml [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+ -->
+
+<smtpservers>
+    <smtpserver enabled="true">
+        <jmxName>smtpserver-global</jmxName>
+        <bind>0.0.0.0:0</bind>
```
Comment (mbaechler) [inline server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/smtpserver.xml:25]: are we using smtp on 3 different ports?

File: server/protocols/jmap-rfc8621-integration-tests/pom.xml [hunk 12]
```diff
@@ -35,6 +35,7 @@
     <modules>
         <module>jmap-rfc8621-integration-tests-common</module>
         <module>memory-jmap-rfc8621-integration-tests</module>
+        <module>distributed-jmap-rfc8621-integration-tests</module>
```
Comment (mbaechler) [inline server/protocols/jmap-rfc8621-integration-tests/pom.xml:38]: you should respect the alphabetic ordering

Comment (nvduc91) [reply server/protocols/jmap-rfc8621-integration-tests/pom.xml:38]: Yes, my bad.

File: server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/imapserver.xml (hunk 8 above)
Comment (nvduc91) [reply server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/imapserver.xml:25]: I dont know yet, i just copy those files when it needed.I will try to understand it.

File: server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/pom.xml (hunk 6 above)
Comment (nvduc91) [reply server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/pom.xml:48]: Yes, it should be `test scope` for deps.

File: server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/imapserver.xml (hunk 8 above)
Comment (chibenwa) [reply server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/imapserver.xml:25]: > are we using imap at all in this project?
> on two different ports?

Yes, because long ago we decided to always start protocols, event the ones we don't use.

(an minor test runtime optimization I proposed was rejected because of this)

Now that is said, my position is unchanged: I would agree to disable protocols we don't use ;-)

File: server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/lmtpserver.xml (hunk 9 above)
Comment (chibenwa) [reply server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/lmtpserver.xml:26]: idem

File: server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/pop3server.xml (hunk 10 above)
Comment (chibenwa) [reply server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/pop3server.xml:25]: idem

File: server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/smtpserver.xml (hunk 11 above)
Comment (chibenwa) [reply server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/smtpserver.xml:25]: idem

File: server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/imapserver.xml (hunk 8 above)
Comment (nvduc91) [reply server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/resources/imapserver.xml:25]: I tried in my computer with:
- imapserver.xml
- lmtpserver.xml
- pop3server.xml
- smtpserver/xml
without content and is breaking nothing. So it not needed, would you agree with it?

**Comment by nvduc91** [comment]
rebase from work at https://github.com/linagora/james-project/pull/3353

File: server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/pom.xml [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+    <parent>
+        <artifactId>jmap-rfc-8621-integration-tests</artifactId>
+        <groupId>org.apache.james</groupId>
+        <version>3.6.0-SNAPSHOT</version>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>distributed-jmap-rfc-8621-integration-tests</artifactId>
+    <description>Distributed Integration testing for Jmap rfc8621</description>
+    <name>Apache James :: Server :: JMAP RFC-8621 :: Distributed Integration Testing</name>
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/pom.xml:32]: LIne breaks please ;-)

File: server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/pom.xml [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>jmap-rfc-8621-integration-tests</artifactId>
+        <groupId>org.apache.james</groupId>
+        <version>3.6.0-SNAPSHOT</version>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>distributed-jmap-rfc-8621-integration-tests</artifactId>
+    <description>Distributed Integration testing for Jmap rfc8621</description>
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/pom.xml:31]: RFC-8621

File: server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/pom.xml [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+                <configuration>
+                    <argLine>-Xms512m -Xmx1024m</argLine>
+                    <reuseForks>true</reuseForks>
+                    <!-- Fail tests longer than 2 hours, prevent form random locking tests -->
+                    <forkedProcessTimeoutInSeconds>7200</forkedProcessTimeoutInSeconds>
+                    <!-- Junit 5 move from Category to Tag, and it need tag's name as parameter -->
+                    <groups combine.self="override">BasicFeature</groups>
+                    <properties>
+                        <includeTags>junit5</includeTags>
+                    </properties>
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/pom.xml:133]: We are missing this configuration Webadmin integration testing and webadmin integration testing.

Can we add this property there too?

File: server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/resources/imapserver.xml [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+
+<imapservers>
+    <imapserver enabled="true">
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/resources/imapserver.xml:24]: Remove servers here?

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EchoMethodContract.scala [hunk 17]
```diff
@@ -130,6 +131,7 @@ trait EchoMethodContract {
   }
 
   @Test
+  @Tag("BasicFeature")
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EchoMethodContract.scala:134]: We should have a 

```
inteface Tags {
   String BASIC_FEATURE = "BasicFeature";
}
```

If we always specify the tag as a string there will be copy & paste mistakes ;-)

File: server/protocols/jmap-rfc-8621-integration-tests/memory-jmap-rfc-8621-integration-tests/pom.xml [hunk 18]
```diff
@@ -28,7 +28,7 @@
 
     <modelVersion>4.0.0</modelVersion>
     <artifactId>memory-jmap-rfc-8621-integration-tests</artifactId>
-
+    <name>Apache James :: Server :: JMAP RFC-8621 :: Memory Integration Testing</name>
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/memory-jmap-rfc-8621-integration-tests/pom.xml:31]: Skip a line after `<name>`

File: server/protocols/jmap-rfc8621-integration-tests/jmap-rfc8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EchoMethodContract.scala [hunk 19]
```diff
@@ -0,0 +1,172 @@
...
+import org.apache.http.HttpStatus
+import org.apache.james.GuiceJamesServer
+import org.apache.james.jmap.JMAPUrls.JMAP
+import org.apache.james.jmap.categories.BasicFeature
+import org.apache.james.jmap.draft.JmapGuiceProbe
+import org.apache.james.jmap.rfc8621.contract.EchoMethodContract._
+import org.junit.experimental.categories.Category
+import org.junit.jupiter.api.{BeforeEach, Test}
+
+object EchoMethodContract {
```
Comment (Arsnael) [inline server/protocols/jmap-rfc8621-integration-tests/jmap-rfc8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EchoMethodContract.scala]: Something is weird... Not sure if it's github loosing control or if it's a rebase issue, but this exact same class is implemented by previous commit from #3353  . The only diff I see is the addition of Category annotations (it's ok, just need to be careful when we rebase after #3353 gets merged)

File: server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/pom.xml [hunk 20]
```diff
@@ -0,0 +1,138 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>apache-james-backends-es</artifactId>
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>apache-james-backends-cassandra</artifactId>
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/pom.xml:42]: wrong order

File: server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/pom.xml [hunk 21]
```diff
@@ -0,0 +1,138 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>com.google.inject.extensions</groupId>
+                    <artifactId>guice-multibindings</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>jmap-rfc-8621-integration-tests-common</artifactId>
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/pom.xml:80]: wrong order

File: server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/pom.xml [hunk 22]
```diff
@@ -0,0 +1,138 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-cassandra-rabbitmq-guice</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-cassandra-guice</artifactId>
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/pom.xml:90]: wrong order

File: server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/pom.xml [hunk 23]
```diff
@@ -0,0 +1,138 @@
...
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-guice-common</artifactId>
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-guice-common</artifactId>
+            <scope>test</scope>
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/pom.xml:109]: not already declared above?

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/pom.xml [hunk 24]
```diff
@@ -28,7 +28,7 @@
 
     <modelVersion>4.0.0</modelVersion>
     <artifactId>jmap-rfc-8621-integration-tests-common</artifactId>
-
+    <name>Apache James :: Server :: JMAP RFC-8621 :: Contract for Integration Testing</name>
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/pom.xml:31]: line break after this line please

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EchoMethodContract.scala (hunk 17 above)
Comment (nvduc91) [reply server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EchoMethodContract.scala:134]: agree

File: server/protocols/jmap-rfc8621-integration-tests/jmap-rfc8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EchoMethodContract.scala (hunk 19 above)
Comment (Arsnael) [reply server/protocols/jmap-rfc8621-integration-tests/jmap-rfc8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EchoMethodContract.scala]: Don't mind this comment (:

File: server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/pom.xml (hunk 15 above)
Comment (nvduc91) [reply server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/pom.xml:133]: I will try it in separate PR

**Comment by nvduc91** [comment]
rebase from work at #3353

File: server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedEchoMethodTest.java (hunk 7 above)
Comment (nvduc91) [reply server/protocols/jmap-rfc8621-integration-tests/distributed-jmap-rfc8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedEchoMethodTest.java]: I dont know why but it needed.

**Comment by nvduc91** [comment]
test this please

**Comment by nvduc91** [comment]
Resolve conflict

File: server/protocols/jmap-rfc8621-integration-tests/jmap-rfc8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EchoMethodContract.scala [hunk 25]
```diff
@@ -0,0 +1,172 @@
...
+      |      "error",
+      |      {
+      |        "type": "Not implemented"
+      |      },
+      |      "notsupport"
+      |    ]
+      |  ]
+      |}""".stripMargin
+
+  private val ACCEPT_RFC8621_VERSION_HEADER: String = """application/json; jmapVersion=rfc-8621"""
```
Comment (remk) [inline server/protocols/jmap-rfc8621-integration-tests/jmap-rfc8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EchoMethodContract.scala]: nitpicking but you don't need triplequotes here

File: server/protocols/jmap-rfc8621-integration-tests/jmap-rfc8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EchoMethodContract.scala [hunk 26]
```diff
@@ -0,0 +1,172 @@
...
+      .setPort(server.getProbe(classOf[JmapGuiceProbe])
+        .getJmapPort
+        .getValue)
+      .setBasePath(JMAP)
+      .build
+  }
+
+  @Test
+  @Category(Array(classOf[BasicFeature]))
+  def echoMethodShouldResponseOKWithRFC8621VersionAndSupportedMethod(): Unit = {
```
Comment (remk) [inline server/protocols/jmap-rfc8621-integration-tests/jmap-rfc8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EchoMethodContract.scala]: s/Response/Respond

File: server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedEchoMethodTest.java [hunk 27]
```diff
@@ -0,0 +1,48 @@
...
+
+    @RegisterExtension
+    static JamesServerExtension testExtension = new JamesServerBuilder()
+        .extension(new DockerElasticSearchExtension())
+        .extension(new CassandraExtension())
+        .extension(new RabbitMQExtension())
+        .extension(new AwsS3BlobStoreExtension())
+        .server(configuration -> GuiceJamesServer.forConfiguration(configuration)
+            .combineWith(CassandraRabbitMQJamesServerMain.MODULES)
+            .overrideWith(TestJMAPServerModule.maximumMessages(LIMIT_TO_20_MESSAGES)))
```
Comment (mbaechler) [inline server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedEchoMethodTest.java:46]: should probably not be overriden

Comment (chibenwa) [reply server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedEchoMethodTest.java:46]: That is done for every jmap test server...

Comment (mbaechler) [reply server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedEchoMethodTest.java:46]: I don't understand, you just proposed a PR to not override the value when not needed. Is this override used for these specific tests?

Comment (chibenwa) [reply server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedEchoMethodTest.java:46]: It is just that this work need a rebase, that is all ;-)

Comment (nvduc91) [reply server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedEchoMethodTest.java:46]: > I don't understand, you just proposed a PR to not override the value when not needed. Is this override used for these specific tests?

i just comment at [here](https://github.com/linagora/james-project/pull/3372#discussion_r423549200)

**Comment by nvduc91** [comment]
resolve conflict

File: server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedEchoMethodTest.java (hunk 27 above)
Comment (mbaechler) [reply server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedEchoMethodTest.java:46]: Well, I suspect that during the rebase/merge we'll keep it by accident.

**Comment by Arsnael** [comment]
```
[96fe4de2abdd44781f6ba734daf2cc7c02548074] [ERROR] /james-project/server/protocols/jmap-rfc-8621-integration-tests/memory-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/memory/MemoryEchoMethodTest.java:[23,1] cannot find symbol
[96fe4de2abdd44781f6ba734daf2cc7c02548074] [ERROR]   symbol:   static LIMIT_TO_20_MESSAGES
[96fe4de2abdd44781f6ba734daf2cc7c02548074] [ERROR]   location: class org.apache.james.modules.TestJMAPServerModule
[96fe4de2abdd44781f6ba734daf2cc7c02548074] [ERROR] /james-project/server/protocols/jmap-rfc-8621-integration-tests/memory-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/memory/MemoryEchoMethodTest.java:[37,64] cannot find symbol
[96fe4de2abdd44781f6ba734daf2cc7c02548074] [ERROR]   symbol:   variable LIMIT_TO_20_MESSAGES
[96fe4de2abdd44781f6ba734daf2cc7c02548074] [ERROR]   location: class org.apache.james.jmap.rfc8621.memory.MemoryEchoMethodTest
```
needs a rebase I think :))

**Comment by nvduc91** [comment]
rebase from master instead #3353

**Comment by chibenwa** [comment]
Merged
